src/app/shared/filter-by-property/filter-by-property.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({
+    name: 'filterByProperty',
+})
+export class FilterByPropertyPipe implements PipeTransform {
+    transform<T>(
+        items: T[] | undefined | null,
+        propertyName: keyof T,
+        searchValue: unknown,
+    ): T[] | undefined | null {
+        if (!Array.isArray(items) || !propertyName || typeof searchValue === 'undefined') {
+            return items;
+        }
+
+        return items.filter(item => {
+            if (typeof item[propertyName] === 'string') {
+                return (item[propertyName] as string)
+                    .toLowerCase()
+                    .includes((searchValue as string).toLowerCase());
```

Comment:
Из-за не строгой типизации я бы подумал в сторону того, чтобы привести значение `searchValue` к строке, а не надеяться на то, что оно будет строкой.

Для примера, предположим, что в свойстве объекта лежит строка `'5'` or `'2'`, и мы передаем число в аргумент пайпа, например `3`. В таком случаи проверка пойдет по ветке проверки строк и у числа будет вызван метод `toLowerCase()`, которого, конечно же, у числа нет, а значит что мы получим? - ошибку

По этому я бы сделал хотя бы самое простое приведение: 
```suggestion
                    .includes(String(searchValue).toLowerCase());
```
OR
```suggestion
                    .includes(searchValue.toString().toLowerCase());
```

Reply:
тут тоже изменила на String(searchValue)